smoketest.bash

```diff
@@ -206,14 +219,30 @@ createProxyCfgVolume() {
     local cfg
     cfg="$(mktemp)"
     chmod 644 "${cfg}"
-    envsubst '$STORAGE_PORT' < "${DIR}/compose/auth_proxy_alpha_config.yaml" > "${cfg}"
+    envsubst '$STORAGE_PORT' < "${DIR}/compose/${AUTH_PROXY_ALPHA_CONFIG_FILE}.yaml" > "${cfg}"
     "${container_engine}" cp "${DIR}/compose/auth_proxy_htpasswd" proxy_cfg_helper:/tmp/auth_proxy_htpasswd
     "${container_engine}" cp "${cfg}" proxy_cfg_helper:/tmp/auth_proxy_alpha_config.yaml
 }
 if [ "${USE_PROXY}" = "true" ]; then
     createProxyCfgVolume
 fi
 
+createProxyCertsVolume() {
+    "${container_engine}" volume create auth_proxy_certs
+    "${container_engine}" container create --name proxy_certs_helper -v auth_proxy_certs:/certs busybox
+    if [ -f "${DIR}/compose/auth_certs/certificate.pem" ] && [ -f "${DIR}/compose/auth_certs/private.key" ]; then
+        chmod 644 "${DIR}/compose/auth_certs/private.key"
+        "${container_engine}" cp "${DIR}/compose/auth_certs/certificate.pem" proxy_certs_helper:/certs/certificate.pem
+        "${container_engine}" cp "${DIR}/compose/auth_certs/private.key" proxy_certs_helper:/certs/private.key
+    else
+        echo "Did you run auth_certs/generate.sh?"
+        exit 2
+    fi
+}
+if [ "${USE_PROXY}" = "true" ]; then
```

Comment:
I think this condition should also test if `USE_HTTPS = true`, so that the certs don't need to have been prepared if insecure HTTP is to be used.

Reply:
I can either leave this check out, or create a new `auth_proxy.yml` file for http without this volume, but I'm more inclined to just do the first option.

Reply:
On the latest commit, without that check, it goes like this:

``bash
$ ./smoktest.bash -Okt
...
+ '[' true = true ']'
+ createProxyCfgVolume
+ podman volume create auth_proxy_cfg
auth_proxy_cfg
+ podman container create --name proxy_cfg_helper -v auth_proxy_cfg:/tmp busybox
939d98b880049a81248681a289e2d9c3917560c49e2cff0c4b3c9018096fa96b
+ local cfg
++ mktemp
+ cfg=/tmp/tmp.ffPxBU0dJ2
+ chmod 644 /tmp/tmp.ffPxBU0dJ2
+ envsubst '$STORAGE_PORT'
+ podman cp /home/work/workspace/cryostat/compose/auth_proxy_htpasswd proxy_cfg_helper:/tmp/auth_proxy_htpasswd
+ podman cp /tmp/tmp.ffPxBU0dJ2 proxy_cfg_helper:/tmp/auth_proxy_alpha_config.yaml
+ '[' true = true ']'
+ createProxyCertsVolume
+ podman volume create auth_proxy_certs
auth_proxy_certs
+ podman container create --name proxy_certs_helper -v auth_proxy_certs:/certs busybox
cfb831e49bb23d6c42ca188f028395b0af0987cc832e29bbd842d24c2b0cde2d
+ '[' -f /home/work/workspace/cryostat/compose/auth_certs/certificate.pem ']'
+ echo 'Did you run auth_certs/generate.sh?'
Did you run auth_certs/generate.sh?
+ exit 2
+ cleanup
+ set +xe
Removing network compose_default
WARNING: Network compose_default not found.
Volume jmxtls_cfg is external, skipping
Volume templates is external, skipping
Removing volume compose_postgresql
WARNING: Volume compose_postgresql not found.
Volume auth_proxy_cfg is external, skipping
Volume auth_proxy_certs is external, skipping
Removing volume compose_seaweed_data
WARNING: Volume compose_seaweed_data not found.
proxy_cfg_helper
proxy_certs_helper
auth_proxy_cfg
auth_proxy_certs
Error: no container with ID or name "jmxtls_cfg_helper" found: no such container
Error: no volume with name "jmxtls_cfg" found: no such volume
Error: no container with ID or name "templates_helper" found: no such container
Error: no volume with name "templates" found: no such volume
```

so I think maybe having a split `auth_proxy.yml` and `auth_proxy_https.yml` compose definition might be needed.

Most of the configuration can just be in `auth_proxy.yml`, and then the `_https` one can just be essentially a patch applied on top that adds in the volume config etc. Then the smoketest script can conditionally add the auth proxy file to the list of compose configs to load, and then conditionally add its HTTPS configs on top of that if enabled.

Reply:
I think it didn't run here because there's no cert in auth_certs?

Reply:
I should make that message clearer

Reply:
I'm trying to use `-k` to avoid needing to have the certs generated - just so that it's very evident that the TLS setup can be fully disabled and everything is working. If the certs are still present and loaded, then it's harder to test and be sure that things are really working with plain HTTP, compared to being able to disable the config and fully remove the certs in which case it's very obvious that it cannot be running HTTPS.